src/fdc/MSXtar.cc

```diff
@@ -265,10 +293,12 @@ unsigned MSXtar::getNextSector(unsigned sector)
 }
 
 // get start cluster from a directory entry,
-// also takes care of BAD_FAT and EOF_FAT-range.
-static unsigned getStartCluster(const MSXDirEntry& entry)
+// BAD FAT and EOF FAT-range are normalized to a single EOF_FAT value (0xFFFFFFF)
+unsigned MSXtar::getStartCluster(const MSXDirEntry& entry) const
 {
-	return normalizeFAT(entry.startCluster);
+	unsigned cluster = entry.startCluster;
+	unsigned badFat = fat16 ? BAD_FAT16 : BAD_FAT12;
+	return cluster < badFat ? cluster : EOF_FAT;
```

Comment:
Should this be `BAD_FAT` or `MAX_FAT`?

Reply:
As per https://en.wikipedia.org/wiki/Design_of_the_FAT_file_system#cite_ref-Ecma-107_5-8, we should be tolerant of reserved cluster number 0xFF6 / 0xFFF6 when reading, and treat it as a normal cluster number. So we do not include it in the normalisation (also not in `readFAT`).

But we should never write it, which is why `writeFAT` asserts the narrower range.

Reply:
Ah OK.
This asymmetrical treatment of 0xFF6 was unexpected to me. Might be worth adding a comment.